src/lib/dev-environment/dev-environment-cli.js

```diff
@@ -95,14 +95,31 @@ const verifyDNSResolution = ( slug: string ) => {
 		} );
 	} );
 };
-
+const VALIDATION_STEPS = [
+	{ id: 'docker', name: 'Check for docker instalation' },
+	{ id: 'compose', name: 'Check for docker-compose instalation' },
+	{ id: 'dns', name: 'Check DNS resolution' },
+];
 export const validateDependencies = async ( slug: string ) => {
+	const progressTracker = new ProgressTracker( VALIDATION_STEPS );
+	console.log( 'Running validation steps...' );
+	progressTracker.startPrinting();
+	progressTracker.stepRunning( 'docker' );
 	try {
 		await validateDockerInstalled();
 	} catch ( exception ) {
 		throw new UserError( exception.message );
 	}
+	progressTracker.stepSuccess( 'docker' );
+	progressTracker.stepSuccess( 'compose' );
+	progressTracker.print();
+
 	await verifyDNSResolution( slug );
+
+	progressTracker.stepSuccess( 'dns' );
```

Comment:
I didn't look into `progressTracker` but I would like to confirm that we don't need

```js
progressTracker.stepRunning( 'dns' );
```

before `await verifyDNSResolution( slug );`